client/packages/programs/src/JsonForms/components/Note.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useEffect, useState } from 'react';
+import { ControlProps, rankWith, uiTypeIs } from '@jsonforms/core';
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import {
+  DetailInputWithLabelRow,
+  useDebounceCallback,
+  useTranslation,
+  useFormatDateTime,
+} from '@openmsupply-client/common';
+import { Typography } from '@mui/material';
+import { FORM_LABEL_WIDTH } from '../common/styleConstants';
+import { z } from 'zod';
+import { useZodOptionsValidation } from '../common/useZodOptionsValidation';
+
+const Options = z
+  .object({
+    width: z.string().optional(),
+    /**
+     * How many rows should the textbox display initially (default: 5)
+     */
+    rows: z.number().optional(),
+  })
+  .strict()
+  .optional();
+
+type Options = z.infer<typeof Options>;
+
+export const noteTester = rankWith(5, uiTypeIs('Note'));
+
+const UIComponent = (props: ControlProps) => {
```

Comment:
This component is mostly the same as the "Text" component, just with an extra line for "Author" field and a different display for when the note is not editable.